File: 01-hello/src/main/java/io/tacsio/rabbitmq/hello/config/RabbitConfig.java
package io.tacsio.rabbitmq.hello.config;

import io.tacsio.rabbitmq.hello.RabbitReceiver;
import io.tacsio.rabbitmq.hello.RabbitSender;
import org.springframework.amqp.core.Queue;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;

@Configuration
public class RabbitConfig {

    @Bean
    public Queue hello() {
        return new Queue("hello");
    }

    @Profile("receiver")
    @Bean
    public RabbitReceiver receiver() {
        return new RabbitReceiver();
    }

    @Profile("sender")
    @Bean
    public RabbitSender sender() {
        return new RabbitSender();
    }
}

File: 07-scheduler/consumer/src/main/kotlin/io/tacsio/udemy/rabbitmq/config/RabbitConfig.kt
package io.tacsio.udemy.rabbitmq.config

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.json.JsonMapper
import org.springframework.amqp.rabbit.config.SimpleRabbitListenerContainerFactory
import org.springframework.amqp.rabbit.connection.ConnectionFactory
import org.springframework.amqp.rabbit.listener.RabbitListenerContainerFactory
import org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer
import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter
import org.springframework.boot.autoconfigure.amqp.SimpleRabbitListenerContainerFactoryConfigurer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile

@Configuration
class RabbitConfig {

    @Bean
    fun objectMapper(): ObjectMapper {
        return JsonMapper.builder().findAndAddModules().build()
    }

    @Bean
    fun converter(objectMapper: ObjectMapper): Jackson2JsonMessageConverter {
        return Jackson2JsonMessageConverter(objectMapper);
    }
}